File: app/src/main/java/com/example/randommusic/mvp/contracts/MainActivityContract.java
package com.example.randommusic.mvp.contracts;

import com.example.randommusic.interfaces.IPresenter;
import com.example.randommusic.interfaces.IView;

public interface MainActivityContract {
    interface View extends IView {
        Boolean hasFragment();
    }

    interface Presenter extends IPresenter<View> {

    }

    interface Model{

    }
}

File: app/src/main/java/com/example/randommusic/mvp/contracts/MainActivityContract.kt
package com.example.randommusic.mvp.contracts

import com.example.randommusic.Action
import com.example.randommusic.interfaces.IPresenter
import com.example.randommusic.interfaces.IView
import com.example.randommusic.interfaces.IMediaCallbacks
import com.example.randommusic.interfaces.IMediaPlayerEvents

class MainActivityContract {
    interface Presenter : IPresenter<View> , IMediaCallbacks {
        fun onAction(action : Action)
        fun onReady(player : IMediaPlayerEvents)
    }

    interface View : IView {
        fun restoreConnection (mediaCallbacks : IMediaCallbacks)
        fun createConnection (mediaCallbacks: IMediaCallbacks)
        fun seekBarTo(position : Int, type : String? = "primary")
    }

    interface Model {

    }
}